backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/extraction/impl/CompositeHitExtractor.java

```diff
@@ -39,9 +39,9 @@ public void contributeFields(Set<String> absoluteFieldPaths) {
 	}
 
 	@Override
-	public void extract(C collector, Document document, Float score) {
+	public void extract(C collector, Document document, int docId, Float score) {
```

Comment:
I went a bit down this road but not entirely: e.g. I introduced `DocumentResult` but I still have a `getDocument()` getter as I was not very excited about passing a `DocumentResult` to the codec.

I can see how introducing a wrapper exposing only the get fields methods of `Document` could be appealing but I suppose we would need another wrapper.

Thoughts?

Reply:
Right, adding a dependency from the codec to the projection package is not a good idea.

Since you're asking: you could declare an interface next to LuceneCodec that defines the operations a codec can expect when extracting information, and make LuceneResult implement that interface. Then you don't have to write much more code (just the one interface), and you don't have to instantiate wrapper objects.

But really, it's fine as it is. We can improve on this later if necessary.